core/src/main/scala/io/finch/endpoint/header.scala

```diff
@@ -11,23 +10,23 @@ private[finch] abstract class Header[F[_], G[_], A](name: String)(implicit
   d: DecodeEntity[A],
   tag: ClassTag[A],
   protected val F: Effect[F]
-) extends Endpoint[F, G[A]] with (Try[A] => Try[Output[G[A]]]) { self =>
+) extends Endpoint[F, G[A]] with (Either[Throwable, A] => Either[Throwable, Output[G[A]]]) { self =>
 
   protected def missing(name: String): F[Output[G[A]]]
   protected def present(value: A): G[A]
 
-  final def apply(ta: Try[A]): Try[Output[G[A]]] = ta match {
-    case Return(a) => Return(Output.payload(present(a)))
-    case Throw(e) => Throw(Error.NotParsed(items.HeaderItem(name), tag, e))
+  final def apply(ta: Either[Throwable, A]): Either[Throwable, Output[G[A]]] = ta match {
+    case Right(a) => Right(Output.payload(present(a)))
+    case Left(e) => Left(Error.NotParsed(items.HeaderItem(name), tag, e))
   }
 
   final def apply(input: Input): EndpointResult[F, G[A]] = {
     val output: F[Output[G[A]]] = F.suspend {
       input.request.headerMap.getOrNull(name) match {
         case null => missing(name)
-        case value => d(value).transform(self) match {
-          case Return(r) => F.pure(r)
-          case Throw(t) => F.raiseError(t)
+        case value => apply(d(value)) match {
```

Comment:
Re `apply`: looks like we don't need to mix-in `Function1` anymore. Any chance you want to make it a simple method instead? Obviously, we can revisit that later in a sense it doesn't need to happen as part of this PR.